ganga/GangaCore/Core/GangaRepository/container_controllers.py

```diff
@@ -173,10 +173,36 @@ def mongod_exists(controller, cname=None):
                 return proc
     return None
 
+
 @repeat_while_none(max=10, message='Waiting for mongod database to start')
 def mongod_exists_wait(controller, cname=None):
     return mongod_exists(controller, cname)
 
+
+def check_sif_file_hash(path):
+    """Check the hash of the sif file used by singularity
+    Args:
+        path (str): Location of the sif file
+    """
+    import hashlib
+
+    DEF_HASH = "a3833a7eb669f21f27de130495e2e1ba"
+
+    m = hashlib.md5()
+    hashfile = open(path, "rb")
+    while True:
```

Comment:
I'm not sure I understand this bit - is it really the best way to read a file?

Reply:
The idea ofthe code here (from StackOverflow) is to avoid having the full file in memory. OK for me.